compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -8134,6 +8140,10 @@ OMR::Z::TreeEvaluator::checkAndSetMemRefDataSnippetRelocationType(TR::Node * nod
       {
       reloType = TR_BlockFrequency;
       }
+   else if (cg->comp()->compileRelocatableCode() && node->getSymbol()->isCatchBlockCounter())
```

Comment:
It's better to replace `compileRelocatableCode()` with something like `needRelocationsForBodyInfoData()`